backend/src/api-token/api-token.service.ts

```diff
@@ -123,30 +128,35 @@ export class ApiTokenService {
       const secretHash = hashApiToken(secret);
       const fullToken = `${AUTH_TOKEN_PREFIX}.${keyId}.${secret}`;
       // Tokens can only be valid for a maximum of 2 years
-      const maximumTokenValidity = new Date();
-      maximumTokenValidity.setTime(
-        maximumTokenValidity.getTime() + 2 * 365 * 24 * 60 * 60 * 1000,
-      );
+      const maximumTokenValidity = getCurrentTimestamp().plus({
+        year: 2,
+      });
       const isTokenLimitedToMaximumValidity =
-        !userDefinedValidUntil || userDefinedValidUntil > maximumTokenValidity;
+        !userDefinedValidUntil ||
```

Comment:
Do we want to make this explicit in case DateTime can evaluate falsy in certain cases?

```suggestion
        userDefinedValidUntil === undefined ||
```